### ISSUE TITLE
Scene Editor and Groups

### ISSUE THREAD

**Issue by Remixful** [boniatillo-com/PhaserEditor#26]
Hello again, @boniatillo-com , you'll probably be seeing me here a lot now! 😀 

So far, the scene editor has proved to be very valuable as I'm creating large worlds with lots of objects, and this could easily result in 300+ lines of code.  I couldn't imagine doing this manually. (I could also make some level editor but I'm too lazy) So far I've been having to use regex to make the generated code cleaner and fit to my tastes. 

One problem I'm having is copying & pasting multiple sprite selections. They do not save the group they were in before. Without this feature, this leaves very tedious work.

![](https://i.gyazo.com/d11e047e2d3d38bd5fe3b50475e05107.gif)

Thank you~

**Comment by boniatillo-com** [comment]
Hi @Remixful please never hesitate to share your issues and ideas. We need feedback!

I don't get very well what's the behavior you expect.

If you want to copy/paste a whole group then you only need to select the group, not the sprites inside it. If you select the sprites, only sprites are copied/pasted. To select a group you have to go to the outline window (the objects tree) and click on it. Maybe in this point we should add a context menu option and key shortcut to select the group of the selected sprite.

But maybe what you need is to pasted the sprites in the same group, right now it is pasting them directly in the root group. It is something we have to implement too, to fix the group where sprites are pasted/dropped. See #14 .

Did I understand well?

**Comment by boniatillo-com** [comment]
By the way, can you tell me a bit about the function of the `tool_clip` and `tool_fringe` sprites? Are them a hidden layer or something like that?

**Comment by Remixful** [comment]
Ah, I see #14 already lists my problem. That's exactly what I'm looking for. I want them to paste and remain in the same group. Sorry for not explaining it well 😞 

I'm pretty much using groups as layers. Here's what they do.
![](https://i.gyazo.com/c7e812e04aba76931abf74043876076e.gif)

**Comment by boniatillo-com** [comment]
> Ah, I see #14 already lists my problem. That's exactly what I'm looking for. I want them to paste and remain in the same group.

Great!

> I will probably just regex the generated code some more to put them into their respective groups until this feature is added.

Some day we are going to open the code generation for plugins. What I usually do is to create the whole scene and just when it is complete I move the objects to their groups. Many times objects with similar names (the name of the texture) are going to the same group so it is just about to filter them in the Outline tree, so you can select them and move to the group. But is true that it needs improvements.

**Comment by boniatillo-com** [comment]
Done, but I implemented it in a different way (tell me what you think). 

Instead of paste in the same group of the cut objects, it pastes on the group of the selected object. But if there is not a selection then it get the group of the copied object.

I did in that way because that rule also apply when you drop an asset, the object is added to the selected group (or the parent of the selection it is not a group).

But for your example it works as you expect, because it will paste on the group of the same object you copied.

The general rule is always select where you want to paste/drop.
